### PR TITLE
Gp2 36 - Mark as complete

### PR DESCRIPTION
**Pull request by lewis-coulson-dit**
To do (delete all that do not apply):

 - [x] Change has a jira ticket that has the correct status.
 - [x] Changelog entry added.
 - [x] (if updating requirements) Requirements have been compiled.

### CODE REVIEW

File: react-components/src/components/MarkLessonAsComplete/MarkLessonAsComplete.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        .then((response) => response.json())
+        .then(({ lesson_completed }) => {
+          if (lesson_completed.length >= 1) {
+            setIsComplete(true)
+          }
+        })
+        .catch((e) => console.log('error: ', e))
+    } else {
+      Services[isComplete ? 'setLessonComplete' : 'setLessonIncomplete'](endpoint).catch((e) =>
+        console.log('error: ', e)
```
Comment (dahfool) [inline react-components/src/components/MarkLessonAsComplete/MarkLessonAsComplete.jsx:22]: Is the `console.log` intentional ?

Comment (lewis-coulson-dit) [reply react-components/src/components/MarkLessonAsComplete/MarkLessonAsComplete.jsx]: good spot, have replaced with empty arrows functions.

File: react-components/src/components/MarkLessonAsComplete/MarkLessonAsComplete.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import React, { useState, useEffect } from 'react'
+import ReactDOM from 'react-dom'
+
+import Services from '../../Services'
+
+const MarkLessonAsComplete = ({ endpoint }) => {
+  const [isComplete, setIsComplete] = useState(undefined)
```
Comment (dahfool) [inline react-components/src/components/MarkLessonAsComplete/MarkLessonAsComplete.jsx:7]: is isComplete a boolean ? If so you can initialised by setting `useState(false)`

Comment (lewis-coulson-dit) [reply react-components/src/components/MarkLessonAsComplete/MarkLessonAsComplete.jsx:7]: Yea I set it to undefined initially so I can do the initial fetch in the `useEffect` hook

Comment (dahfool) [reply react-components/src/components/MarkLessonAsComplete/MarkLessonAsComplete.jsx:7]: Does it have 3 states, `undefined`, 'false' and 'true'

Comment (dahfool) [reply react-components/src/components/MarkLessonAsComplete/MarkLessonAsComplete.jsx:7]: 👍

Comment (lewis-coulson-dit) [reply react-components/src/components/MarkLessonAsComplete/MarkLessonAsComplete.jsx:7]: I can add a different var like `hasLoaded` I guess, that just seemed like a nice way of differentiating from the initial load state and the toggle states.

**Pull request merged**